packages/api/tests/es/test-collections.js

```diff
@@ -0,0 +1,361 @@
+
+  esAlias = randomId('esalias');
+  esIndex = randomId('esindex');
+  process.env.ES_INDEX = esAlias;
+
+  // create the elasticsearch index and add mapping
+  await bootstrapElasticSearch('fakehost', esIndex, esAlias);
+  esClient = await Search.es();
+
+  await Promise.all([
+    indexer.indexCollection(esClient, fakeCollectionFactory({
+      name: 'coll1',
+      version: '1'
+    }), esAlias),
+    indexer.indexCollection(esClient, fakeCollectionFactory({
+      name: 'coll1',
+      version: '2'
+    }), esAlias),
+    indexer.indexCollection(esClient, fakeCollectionFactory({
+      name: 'coll2',
+      version: '1'
+    }), esAlias),
+    indexer.indexGranule(esClient, fakeGranuleFactoryV2({ collectionId: 'coll1___1' }), esAlias),
+    indexer.indexGranule(esClient, fakeGranuleFactoryV2({ collectionId: 'coll1___1' }), esAlias)
+  ]);
+
+  // Indexing using Date.now() to generate the timestamp
+  const stub = sinon.stub(Date, 'now').returns((new Date(2020, 0, 29)).getTime());
+
+  await Promise.all([
+    indexer.indexCollection(esClient, fakeCollectionFactory({
+      name: 'coll3',
+      version: '1',
+      updatedAt: new Date(2020, 0, 29)
+    }), esAlias),
+    indexer.indexGranule(esClient, fakeGranuleFactoryV2({
+      updatedAt: new Date(2020, 1, 29),
+      collectionId: 'coll3___1'
+    }), esAlias)
+  ]);
+
+  stub.restore();
+});
+
+
+test.after.always(async () => {
+  await esClient.indices.delete({ index: esIndex });
+  await collectionModel.deleteTable();
+  await granuleModel.deleteTable();
+  await s3.recursivelyDeleteS3Bucket(process.env.system_bucket);
+});
+
+test.serial('getStats returns an empty list if there are no records or ids', async (t) => {
+  const collectionSearch = new Collection({}, null, process.env.ES_INDEX);
+  const stats = await collectionSearch.getStats([], []);
+
+  t.deepEqual([], stats);
+});
+
+test.serial('getStats returns an empty list if there are no records', async (t) => {
+  const collectionSearch = new Collection({}, null, process.env.ES_INDEX);
+  const stats = await collectionSearch.getStats([], ['coll1___1']);
+
+  t.deepEqual(stats, []);
+});
+
+test.serial('getStats returns empty stats if there are no ids', async (t) => {
+  const records = [
+    { name: 'coll1', version: '1' },
+    { name: 'coll1', version: '2' }
+  ];
+  const collectionSearch = new Collection({}, null, process.env.ES_INDEX);
+  const stats = await collectionSearch.getStats(records, []);
+
+  t.deepEqual(stats,
+    records.map((r) => ({
+      ...r,
+      stats: {
+        running: 0,
+        completed: 0,
+        failed: 0,
+        total: 0
+      }
+    })));
+});
+
+test.serial('getStats correctly adds stats', async (t) => {
+  const collectionSearch = new Collection({}, null, process.env.ES_INDEX);
+  const stats = await collectionSearch.getStats([{ name: 'coll1', version: '1' }], ['coll1___1']);
+
+  t.is(stats.length, 1);
+  t.deepEqual(stats[0].stats, {
+    running: 0,
+    completed: 2,
```

Comment:
So this works because `fakeGranuleFactoryV2` generates a granule with a `completed` status by default? Should we make that explicit for the test?

Reply:
I updated it to make it easier to understand